core/bot_framework_adapter_test.go

```diff
@@ -20,18 +20,58 @@
 package core_test
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
 	"fmt"
-	"log"
+	"net/http"
+	"net/http/httptest"
 	"os"
+	"testing"
 
 	"github.com/infracloudio/msbotbuilder-go/core"
 	"github.com/infracloudio/msbotbuilder-go/core/activity"
 	"github.com/infracloudio/msbotbuilder-go/schema"
+	"github.com/stretchr/testify/assert"
 )
 
-func Example() {
+func serverMock() *httptest.Server {
+	handler := http.NewServeMux()
+	handler.HandleFunc("/v3/conversations/abcd1234/activities", msTeamsMockMock)
 
+	srv := httptest.NewServer(handler)
+
+	return srv
+}
+
+func msTeamsMockMock(w http.ResponseWriter, r *http.Request) {
+	_, _ = w.Write([]byte("{\"id\":\"1\"}"))
+}
+
+// Create a handler that defines operations to be performed on respective events.
+// Following defines the operation to be performed on the 'message' event.
+var customHandler = activity.HandlerFuncs{
+	OnMessageFunc: func(turn *activity.TurnContext) (schema.Activity, error) {
+		return turn.SendActivity(activity.MsgOptionText("Echo: " + turn.Activity.Text))
+	},
+}
+
+func processMessage(w http.ResponseWriter, req *http.Request) {
+	ctx := context.Background()
+	setting := core.AdapterSetting{
+		AppID:       "asdasd",
+		AppPassword: "cfg.MicrosoftTeams.AppPassword",
+	}
+	adapter, err := core.NewBotAdapter(setting)
```

Comment:
Fail the test if err != nil

Reply:
@PrasadG193 , processMessage function is a stub which is performing a behaviour and same is asserted in test case. It is better not to put asserts in stubbing code.

Reply:
@coupa-sbawaskar how do we validate if the functions then? What would you suggest?

Reply:
Function web response is asserted in test. There we can add asserts. This particular function will be user implemented code and not part of framework. So its okay not to test its behaviour.

Reply:
Got your point, please check updated code. Mocked token validator so that auth is bipassed.